src/Dialect/ONNX/ElementsAttr/ElementsAttrBuilder.cpp

```diff
@@ -517,6 +517,120 @@ ElementsAttr ElementsAttrBuilder::reduce(ElementsAttr elms,
   });
 }
 
+ElementsAttr ElementsAttrBuilder::matMul(ElementsAttr lhs, ElementsAttr rhs) {
+  ShapedType lhsType = lhs.getType();
+  ShapedType rhsType = rhs.getType();
+  Type elementType = lhsType.getElementType();
+  assert(elementType == rhsType.getElementType() &&
+         "matMul() requires identical element types");
+
+  ArrayRef<int64_t> lhsShape = lhsType.getShape();
+  unsigned lhsRank = lhsShape.size();
```

Comment:
Do we use `unsigned` in other places in the codebase?

Reply:
MLIR uses unsigned in many places, e.g. for indexes into shapes and operands and results lists

maybe it's a little bit unusual to use it for rank? I changed it to size_t